local-cli/generator-android/templates/src/app/build.gradle

```diff
@@ -127,7 +127,7 @@ android {
 
 dependencies {
     compile fileTree(dir: "libs", include: ["*.jar"])
-    compile "com.android.support:appcompat-v7:23.0.1"
+    compile "com.android.support:appcompat-v7:24.2.0"
```

Comment:
Updating appcompat version is not that easy.
RN is coupled with 23.0.1 right now because we have Java code in RN that is coupled to specific versions of its dependencies, see my [explanation](https://github.com/facebook/react-native/pull/9340).

The best way to move forward is if someone abstracts out RN dependencies so that users could provide/switch implementations during build times.
Help is wanted and needed.